hibernate-core/src/main/java/org/hibernate/event/internal/WrapVisitor.java

```diff
@@ -104,10 +108,31 @@ final Object processArrayOrNewCollection(Object collection, CollectionType colle
 					if ( attributeInterceptor instanceof EnhancementAsProxyLazinessInterceptor ) {
 						return null;
 					}
+					else if ( attributeInterceptor != null
+							&& !session.getFactory().getSessionFactoryOptions().isCollectionsInDefaultFetchGroupEnabled() ) {
```

Comment:
I think it would be more correct to do the following instead:

```suggestion
							&& !attributeInterceptor.isAttributeLoaded( persister.getNavigableRole().getLocalName() ) ) {
```

Reply:
you mean `&& ((LazyAttributeLoadingInterceptor)attributeInterceptor).isAttributeLoaded( persister.getAttributeMapping().getAttributeName() )`?